### PR TITLE
made it madatory to accept the privacy policy

### LINKED ISSUE #650: Require accepting privacy policy
Require the accepting of a privacy policy on google login
Dispense of right of withdrawal on purchase.

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #650

### CODE REVIEW

File: components/GoogleSignIn/GoogleSignIn.tsx [hunk 1]
```diff
@@ -56,7 +59,7 @@ function GoogleSignIn(props: Props) {
         setIsLoggedIn(true)
         api.setGoogle(response.tokenId)
             .then(() => {
-                (window as any).googleAuthObj = response
+                ;(window as any).googleAuthObj = response
```
Comment (Ekwav) [inline components/GoogleSignIn/GoogleSignIn.tsx:61]: Why is the ; at the start?

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: components/GoogleSignIn/GoogleSignIn.tsx (hunk 1 above)
Comment (matthias-luger) [reply components/GoogleSignIn/GoogleSignIn.tsx:61]: https://stackoverflow.com/questions/7145514/whats-the-purpose-of-starting-semi-colon-at-beginning-of-javascript

Formatter wants it that way

**Review by Ekwav — CHANGES_REQUESTED** [review]
doesn't build anymore due to merge conflicts

File: components/Premium/Premium.tsx [hunk 2]
```diff
@@ -138,6 +138,7 @@ function Premium() {
                             </Modal.Body>
                         </Modal>
                     </h2>
+                    <p>By buying one of the following products, you confirm the immediate execution of the contract, hereby losing your cancellation right.</p>
```
Comment (Ekwav) [inline components/Premium/Premium.tsx]:
```suggestion
                    <p>By purchasing one of the following products, you confirm the immediate execution of the contract and thus lose your right of withdrawal.</p>
```
I am pretty sure that the user has to explicitly confirm that he want it to execute immediately (via a checkbox) although I think the confirmation can be saved and doesn't have to be repeated

**Comment by Flou21** [comment]
@matthias-luger compile fehler

File: components/CoflCoins/CoflCoinsPurchase.tsx [hunk 3]
```diff
@@ -20,7 +24,7 @@ export default function CoflCoinsPurchase() {
         <div className="cofl-coins-purchase">
             {!isLoggedIn ? <p>To buy and use CoflCoins you need to be logged in with Google</p> : ''}
             <GoogleSignIn onAfterLogin={onLogin} onLoginFail={onLoginFail} />
-            {isLoggedIn ? <Payment /> : ''}
+            {isLoggedIn ? <Payment disabled={props.disabled} /> : ''}
```
Comment (Ekwav) [inline components/CoflCoins/CoflCoinsPurchase.tsx:27]: why would the login be disabled?

File: components/Premium/Premium.tsx (hunk 2 above)
Comment (Ekwav) [reply components/Premium/Premium.tsx]: oh, I am sorry apparently this wasn't saved with my last review. Explains why you were so confused :neutral_face:

File: components/CoflCoins/CoflCoinsPurchase.tsx (hunk 3 above)
Comment (matthias-luger) [reply components/CoflCoins/CoflCoinsPurchase.tsx:27]: The login is not disabled?
The disabled prop is passed to the Payment-Component to disable the Payment-Buttons

**Review by Ekwav — APPROVED** [review]

**Pull request merged**